src/lib.rs

```diff
@@ -169,6 +106,30 @@ impl Default for MaybeInitializedCaptureIndex {
 
 const ALL_NODES_QUERY: &str = "(_) @node";
 
+#[derive(Default)]
+struct CachedQueries(Mutex<HashMap<SupportedLanguageName, Option<Arc<Query>>>>);
```

Comment:
Wanted at least an alias for the long `Mutex<...>` type but then a couple things felt like they could nicely be methods on it so made it a newtype wrapper 

It seems like there are pluses/minuses to "concealing" the fact that this is/contains a `Mutex`?

Reply:
I'll look at the rest of this PR more closely tomorrow, but this line jumped out at me as something that could be replaced either by a concurrent hashmap (eg https://crates.io/crates/dashmap) or by a `RwLock` around the hashmap. It maybe is not worth pulling in a dependency for this to get a real concurrent hashmap, but even a `RwLock` would probably be an improvement (it only locks when writing, so you can have unlimited concurrent readers).

Reply:
Ok I think I'd commented in the first PR where I added this data structure that "really there should probably be a `Mutex` per-language to be more optimal", so I guess along those lines something "clicked" - we don't actually need a `Mutex`, since (for each language) this is a "lazy-initialize and then just read" pattern, aka (since this is being accessed across multiple threads) a `OnceLock`

So updated to use that and it did indeed compile/work, in order to do that (or similarly the per-language `Mutex` that I'd originally mentioned) you do need to actually initialize all of the potential `HashMap` language "entries" at creation time (or else you'd still need shared (cross-thread) mutable access to the `HashMap` itself and would be back to needing a top-level `Mutex`)

(or to test my understanding say that we didn't want to initialize them all or didn't know all of the potential keys, that's where something like a concurrent hashmap would be an alternative to/avoid needing a top-level `Mutex`?)

Reply:
> (or to test my understanding say that we didn't want to initialize them all or didn't know all of the potential keys, that's where something like a concurrent hashmap would be an alternative to/avoid needing a top-level Mutex?)

Yeah, I think so, and you'd probably still keep the inner `OnceLock`s inside that concurrent hashmap so you're really only locking when necessary?